app/helpers/declarable_helper.rb

```diff
@@ -1,4 +1,34 @@
 module DeclarableHelper
+  def declarable_stw_link(declarable, search)
+    geographical_area = GeographicalArea.find(search.country)
+    declarable_type = declarable.heading? ? 'heading' : 'commodity'
+    today = Time.zone.today
+
+    stw_options = {
+      commodity: declarable.code,
+      originCountry: search.country,
+      goodsIntent: 'bringGoodsToSell',
+      userTypeTrader: 'true',
+      tradeType: 'import',
+      destinationCountry: 'GB',
+      importDeclarations: 'yes',
+      importOrigin: nil,
+      importDateDay: search.day.presence || today.day,
+      importDateMonth: search.month.presence || today.month,
+      importDateYear: search.year.presence || today.year,
+    }
+
+    stw_link = "#{TradeTariffFrontend.single_trade_window_url}?#{CGI.unescape(stw_options.to_query)}"
+
+    link_to(
+      "Check how to import #{declarable_type} #{declarable.code} from #{geographical_area&.description}.",
+      stw_link,
+      target: '_blank',
+      class: 'govuk-link',
+      rel: 'noopener',
+    )
+  end
```

Comment:
yes, it's a bit long, but it is readable, there is just a conditional (the ternary op)